apps/functions/dns-redirecting/index.ts

```diff
@@ -42,6 +43,17 @@ export const dnsRedirecting = functions.https.onRequest(
     } else if (hostname === 'blog.angular.io') {
       response.redirect(redirectType, `https://blog.angular.dev${request.originalUrl}`);
     } else if (hostname === 'material.angular.io') {
+      if (request.originalUrl === '/assets/versions.json') {
+        const versionsData = await fetch('https://material.angular.dev/assets/versions.json').then(
+          (r) => r.json(),
+        );
+        const cacheDuration = 60 * 60 * 12;
```

Comment:
In `firebase.json` we have:

```
            "key": "Cache-Control",
            // The versions file should not be cached as it may change
            // at any time and otherwise old versions would be rendered.
            "value": "no-cache
```

Should we do that here as well? I don't think we'd want to cache?

Reply:
My thinking was that we did want to cache this since it is called on literally every first call of a m.aio page. Since this is supposed to be the redirect service I just didn't want it to cause blocking too often.

Reply:
Okay, I'll do no cache for now and I will check in on if we see queuing or overly long requests for the dns redirection service and we can revisit if that comes up.